extensions/amp-copy/0.1/amp-copy.js

```diff
@@ -0,0 +1,180 @@
+
+/** @const */
+const IOS_SELECT = 'Tap the selected text to copy.';
+
+/** @const */
+const COPY_ERROR = 'This browser does not support copying. Please select the text manually, and copy';
+
+class AmpCopy extends AMP.BaseElement {
+
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    /**
+     * @private {?Attribute}
+     */
+    this.copyText_ = null;
+
+    /**
+     * @private {?Element}
+     */
+    this.displayedText_ = null;
+
+    /**
+     * @private {?Element}
+     */
+    this.childContainer_ = null;
+
+    /**
+     * @private {?Element}
+     */
+    this.copyBtn_ = null;
+
+    /**
+     * @private {?Element}
+     */
+    this.copyNotification_ = null;
+
+    /**
+     * @private {?Boolean}
+     */
+    this.isIos_ = null;
+
+    /**
+     * @private {?Timeout}
+     */
+    this.currentNotificationTimeout_ = null;
+  }
+
+  /** @override */
+  buildCallback() {
+    //Get our copy text attribute, and assert its existance
+    const copyTextAttr = user().assert(this.element.getAttribute('copy-text'),
+        'The copy-text attribute is required. %s', this.element);
+    this.copyText_ = copyTextAttr;
+
+    //Create the displayed text element, and assert its existance
+    const textElementAttr = user().assert(this.element.getAttribute('text-element'),
+        'The text-element attribute is required. %s', this.element);
+    //Ensure the text element is supported by the copy
+    user().assert(textElementAttr === 'textarea' || textElementAttr === 'input',
+        'The text-element attribute must be, "input" or a "textarea". %s', this.element);
+    this.displayedText_ = this.element.ownerDocument.createElement(textElementAttr);
+    this.displayedText_.className = "amp-copy-" + textElementAttr;
+    //Setting read-only, to make the text only selectable
+    this.displayedText_.setAttribute('readonly', '');
+    this.displayedText_.value = this.copyText_;
+
+    //Create a container for our copy-button and notification
+    this.childContainer_ = this.element.ownerDocument.createElement('div');
+    this.childContainer_.className = "amp-copy-child-container";
+
+    //Get if we are currently on ios
+    this.isIos_ = navigator.userAgent.match(IOS_USER_AGENT);
+
+    //Create the Copy Button element
+    this.copyBtn_ = this.element.ownerDocument.createElement('button');
+    this.copyBtn_.className = "amp-copy-button";
+    this.copyBtn_.addEventListener('click', () => this.copyBtnClick_());
+    //Set the text dynamically based if it is mobile safari or not.
+    //Mobile Safari can only select the text,
+    //and not copy it like all other browsers
+    if(this.isIos_) {
+      this.copyBtn_.textContent = 'Select';
+    } else {
+      this.copyBtn_.textContent = 'Copy';
+    }
+
+    //Add the created the elements
+    this.element.appendChild(this.displayedText_);
+    this.element.appendChild(this.childContainer_);
+    this.childContainer_.appendChild(this.copyBtn_)
+  }
+
```

Comment:
Please see Line 175.

Unless you want me to move it up there, which I definitely can :)